src/kvasir/mpl/algorithm/flatten.hpp

```diff
@@ -27,7 +27,9 @@ namespace kvasir {
 			};
 
 			template <template <class...> class L, class... Ts>
-			struct flatten_impl<L<Ts...>> : flatten_element_impl<L, L<Ts...>> {};
+			struct flatten_impl<L<Ts...>> {
+				using type = flatten_element_impl<L, L<Ts...>>;
```

Comment:
yes, this is approximately the exact situation I tested while evaluating the difference in performance between inheritance and aliases:
```cpp
template<unsigned n>
struct count : count<n - 1> {};
template<>
struct count<0> {
    using type = void;
}
```
versus
```cpp
template<unsigned n>
struct count {
    using type = typename count<n - 1>::type;
};
template<>
struct count<0> {
    using type = void;
}
```
which is several times faster than the inheritance based counterpart